### ISSUE TITLE
Lint/NestedMethodDefinition should not be triggered in {class|instance|module}_exec

### ISSUE THREAD

**Issue by lordofthelake** [bbatsov/rubocop#3691]
This code doesn't result in a violation:

```ruby
Module.new.class_eval do
  def foo
  end
end    
```

While this does:
```ruby
Module.new.class_exec do
  def foo
  end
end    
```

Since `class_eval` and `class_exec` are semantically equivalent, with the only difference of being able to pass context through `class_exec`, the cop behaviour should be the same in both cases.

## RuboCop version

```
$ rubocop -V
0.43.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by mikegee** [comment]
I had to wrap the examples given in methods to duplicate the problem. For instance:

```ruby
def make_module
  Module.new.class_eval do
    def foo
    end
  end
end
```

**Comment by tejasbubane** [comment]
@mikegee I could reproduce on wrapping `class_exec` inside a method and not `class_eval`.

**Comment by Drenmi** [comment]
This was fixed in #4212. Ping @jonas054. @bbatsov. 🙂
